docs(legal) create copyright file

coopr opened the pull request:
This file is a concatenation of all the licenses and copyright notices from all libraries that are used in Kong CE and EE. As such, some notices in this file may not apply to this Kong CE repo, as they cover elements present only in Kong EE.

thibaultcha commented:
Thanks @coopr, very much appreciated!!

> As such, some notices in this file may not apply to this Kong CE repo, as they cover elements present only in Kong EE.

That is unfortunate, because I do not think those should be included in this copyright file.

coopr commented:
Roger that @thibaultcha - I'm preparing a CE-only COPYRIGHT file, and will revise this PR soon...

rainest commented:
Replaced COPYRIGHT with version that only includes CE libraries.

thibaultcha commented on COPYRIGHT:
```diff
@@ -0,0 +1,1865 @@
+%%%%%%%%%
+
+kong-lapis
+https://github.com/leafo/lapis
```
It seems like we are lacking the license for Lapis (should be MIT)

coopr replied:
The lapis repo https://github.com/leafo/lapis does not have a LICENSE file or other statement of license that allows Github's API to return the contents of lapis' license. This COPYRIGHT file is generated programmatically, and such automation depends on each library's license content being retrievable via a URL (eg. pgmoon's license can is available at https://raw.githubusercontent.com/leafo/pgmoon/master/LICENSE)

thibaultcha commented on COPYRIGHT:
```diff
@@ -0,0 +1,1865 @@
+          of the NOTICE file are for informational purposes only and
+          do not modify the License. You may add Your own attribution
+          notices within Derivative Works that You distribute, alongside
+          or as an ad
```
We do not ship CE with libnettle

coopr replied:
will be fixed in the next version of COPYRIGHT

thibaultcha commented on COPYRIGHT:
```diff
@@ -0,0 +1,1865 @@
+          of the NOTICE file are for informational purposes only and
+          do not modify the License. You may add Your own attribution
+          notices within Derivative Works that You distribute, alongside
+          or as an ad
```
We do not ship CE with lua-resty-nettle

coopr replied:
will be fixed in the next version of COPYRIGHT

thibaultcha commented on COPYRIGHT:
```diff
@@ -0,0 +1,1865 @@
+          of the NOTICE file are for informational purposes only and
+          do not modify the License. You may add Your own attribution
+          notices within Derivative Works that You distribute, alongside
+          or as an ad
```
We do not ship CE with lua-resty-session afaik?

coopr replied:
will be fixed in the next version of COPYRIGHT

thibaultcha commented on COPYRIGHT:
```diff
@@ -0,0 +1,1865 @@
+          of the NOTICE file are for informational purposes only and
+          do not modify the License. You may add Your own attribution
+          notices within Derivative Works that You distribute, alongside
+          or as an ad
```
I believe we only ship libgmp in EE, along with nettle?

coopr replied:
will be fixed in the next version of COPYRIGHT

thibaultcha commented on COPYRIGHT:
```diff
@@ -0,0 +1,1865 @@
+          of the NOTICE file are for informational purposes only and
+          do not modify the License. You may add Your own attribution
+          notices within Derivative Works that You distribute, alongside
+          or as an ad
```
This should probably be removed

coopr replied:
I've removed libraries that are from https://github.com/Kong/

thibaultcha commented on COPYRIGHT:
```diff
@@ -0,0 +1,1865 @@
+          of the NOTICE file are for informational purposes only and
+          do not modify the License. You may add Your own attribution
+          notices within Derivative Works that You distribute, alongside
+          or as an ad
```
None of these have their copyright and licence notices :/

coopr replied:
They weren't found with a quick search. Now must occur a slower search...

coopr replied:
I'm not sure which lines you are referring to - I think Github's web UI is showing me your comment in the wrong place

hishamhm commented:
If we're adding runtime dependencies like Nginx/OpenResty/Lua to the list, then it needs to include LuaRocks as well (it's not only a build dependency, as the Kong code does `require("luarocks.loader")`)

thibaultcha commented:
@hishamhm Good point - LuaRocks was indeed in the list I handed off to @coopr a few days ago; forgot about it here.

coopr commented:
I'm not sure how LuaRocks got dropped from the list - I've added it back, and added https://github.com/luarocks/luarocks/issues/829 to make LuaRocks' licensing more "normal"

thibaultcha commented on COPYRIGHT:
```diff
@@ -0,0 +1,1090 @@
+
+%%%%%%%%%
+
+Open-Resty
```
Could we spell OpenResty correctly? :/

coopr replied:
That is how the library identifies itself, it seems.

hishamhm replied:
"OpenResty®", which is how they call it in every single use of the name in their front page :laughing:

thibaultcha replied:
It is registered since last year or so, after all :)

thibaultcha replied:
Where? I could not find any occurence of OpenResty spelled "Open-Resty". This is certainly not how it is generally spelled (in fact, I have never seen it spelled that way, ever). The copyright is in fact attributed to "Yichun "agentzh" Zhang, OpenResty Inc." (no dashes).

thibaultcha commented on COPYRIGHT:
```diff
@@ -0,0 +1,1090 @@
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
```
Our lua-resty-mediador library is a fork from https://github.com/simbiose/mediador of which we are missing the license (MIT)

lpack (on top of which our lua_pack library is implemented) is in the public domain, so nothing to do here I believe.

coopr replied:
Sorry, I'm unclear if you are requesting a change or not @thibaultcha thanks.

thibaultcha replied:
imho, we should have the following copyright for lua-resty-mediador:
```
Copyright (c)
    2015 simbiose
    2017-2018 Kong, Inc.
```

Or in the current form given by the [LICENSE file](https://github.com/Kong/lua-resty-mediador/blob/master/LICENSE):
```
Copyright (c) 2015 simbiose, 2017 Mashape, Inc.
```

(But slightly outdated)

lua_pack seems to be :ok_hand:

thibaultcha replied:
And I believe that would be the last update before this be merged! We can still revisit later if we forgot something.

thibaultcha commented:
Did not have much time today because of rc2, but will take a final look at this tomorrow!

coopr commented:
Confirmed, this is in an acceptable state for merging.
